### PR TITLE
feat: [CDS-36176]: delegate selector support at advance tab in deploy and pipleine level

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by kapilkumawat86 — APPROVED** [review]

**Comment by dhruvUpadhyay07** [comment]
trigger cypress

**Review by samarth-sharma31 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployAdvancedSpecifications/__test__/__snapshots__/DeployAdvancedSpecifications.test.tsx.snap [hunk 1]
```diff
@@ -8,6 +8,127 @@ exports[`Deploy advanced specifications test Should match snapshot 1`] = `
...
+                              <div
+                                class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
+                              >
+                                <div
+                                  class="bp3-tag-input-values"
+                                >
+                                  <input
+                                    class="bp3-input-ghost bp3-multi-select-tag-input-input"
+                                    placeholder="delegate.Delegate_Selector_placeholder"
+                                    value=""
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployAdvancedSpecifications/__test__/__snapshots__/DeployAdvancedSpecifications.test.tsx.snap:113]: Same as other comment regarding **name**

**Review by chetanvekariya-harness — APPROVED** [review]
Please check comments. If applicable, fix those.

File: src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/__snapshots__/StageAdvancedInputSetForm.test.tsx.snap [hunk 2]
```diff
@@ -15,6 +15,113 @@ exports[`StageAdvancedInputSetForm tests viewType DeploymentForm initial render
...
+                            class=""
+                          >
+                            <div
+                              class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
+                            >
+                              <div
+                                class="bp3-tag-input-values"
+                              >
+                                <input
+                                  class="bp3-input-ghost bp3-multi-select-tag-input-input"
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/__snapshots__/StageAdvancedInputSetForm.test.tsx.snap:109]: Could you please check if **name** field should be here?

File: src/modules/70-pipeline/components/PipelineStudio/AdvancedOptions/__test__/__snapshots__/AdvancedOptions.test.tsx.snap [hunk 3]
```diff
@@ -240,6 +240,128 @@ exports[`<AdvancedOptions/> tests should match snapshot 1`] = `
...
+                              class=""
+                            >
+                              <div
+                                class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
+                              >
+                                <div
+                                  class="bp3-tag-input-values"
+                                >
+                                  <input
+                                    class="bp3-input-ghost bp3-multi-select-tag-input-input"
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/AdvancedOptions/__test__/__snapshots__/AdvancedOptions.test.tsx.snap:347]: Same as other **name** related comments.

File: src/modules/70-pipeline/components/PipelineStudio/DelegateSelector/__tests__/__snapshots__/DelegateSelector.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                                <path
+                                  d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
+                                  fill-rule="evenodd"
+                                />
+                              </svg>
+                            </span>
+                          </button>
+                        </span>
+                        <input
+                          class="bp3-input-ghost bp3-multi-select-tag-input-input"
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/DelegateSelector/__tests__/__snapshots__/DelegateSelector.test.tsx.snap:122]: Same as other comment regarding **name**

File: src/modules/70-pipeline/components/PipelineStudio/DelegateSelector/__tests__/__snapshots__/DelegateSelector.test.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+                    class=""
+                  >
+                    <div
+                      class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
+                    >
+                      <div
+                        class="bp3-tag-input-values"
+                      >
+                        <input
+                          class="bp3-input-ghost bp3-multi-select-tag-input-input"
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/DelegateSelector/__tests__/__snapshots__/DelegateSelector.test.tsx.snap:224]: Same as other comment regarding **name**

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStep.tsx [hunk 6]
```diff
@@ -72,6 +72,10 @@ export class StepGroupStep extends PipelineStep<StepGroupElementConfig> {
     )
   }
 
+  constructor() {
+    super()
+    this._hasDelegateSelectionVisible = true
```
[the remark this answers is not in the document]
Comment (dhruvUpadhyay07) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStep.tsx:77]: this change was asked by kapil

**Pull request merged**